signature/src/mechanism/cosign/README.md

```diff
@@ -0,0 +1,206 @@
+blob.
+
+A manifest's format may look like
+```json
+{
+   "schemaVersion": 2,
+   "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+   "config": {
+      "mediaType": "application/vnd.docker.container.image.v1+json",
+      "size": 1472,
+      "digest": "sha256:d7d3d98c851ff3a95dbcb70ce09d186c9aaf7e25d48d55c0f99aae360aecfd53"
+   },
+   "layers": [
+      {
+         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
+         "size": 2798806,
+         "digest": "sha256:530afca65e2ea04227630ae746e0c85b2bd1a179379cbf2b6501b49c4cab2ccc"
+      }
+   ]
+}
+```
+
+Here the layers are the components of the image, and can be addressed
+by the `digest` field in the blob.
+
+The manifest can be addressed due to the image's reference.
+
+To avoid extra storage for a signature, Cosign smartly store the signature
+and `Payload` in the same registry as the image.
+
+#### "image" format
+
+As the example, when the signature of `172.17.0.3:5000/alpine:latest` is stored,
+there will be a new "image" to be stored, which "image" contains
+the signature the the signed `Payload`. Here the "image" is not a
+true image, but some files following the [OCI Distribution Spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md).
+
+The manifest of the "image" will include the signature encoded in base64, and
+have a pointer to the `Payload`. May look like the following
+
+```json
+{
+    "schemaVersion": 2,
+    "mediaType": "application/vnd.oci.image.manifest.v1+json",
+    "config": {
+        "mediaType": "application/vnd.oci.image.config.v1+json",
+        "size": 233,
+        "digest": "sha256:eca5f5f99822d62181a93ebe1df31db7eeebdf47c47b0a5515c26adc3abf3226"
+    },
+    "layers": [
+        {
+            "mediaType": "application/vnd.dev.cosign.simplesigning.v1+json",
+            "size": 238,
+            "digest": "sha256:a252c3b3b26bad13a82cb65b252d5f409ab72d7db433ceb7242ebdbcf0fd9889",
+            "annotations": {
+                "dev.cosignproject.cosign/signature": "MEUCIQDVshyS+xIpdq5YkAAvrACS5wzg3Zsk4n4UTwcgjiwsnwIgfhy+SeAocHKQVFjX4HgljoAOu27g/p4E2x8izHicbeY="
+            }
+        }
+    ]
+}
+```
+
+Here, `"dev.cosignproject.cosign/signature"` in `"annotations"` is the base64 encoded
+signature for the image using ECDSA P256 algorithm.
+
+The layer `sha256:a252c3b3b26bad13a82cb65b252d5f409ab72d7db433ceb7242ebdbcf0fd9889` in
+the blob, is the `Payload`. It is exactly the example `Payload` mentioned before.
+
+```json
+{
+    "critical": {
+        "identity": {
+            "docker-reference": "172.17.0.3:5000/alpine"
+        },
+        "image": {
+            "docker-manifest-digest": "sha256:9b2a28eb47540823042a2ba401386845089bb7b62a9637d55816132c4c3c36eb"
+        },
+        "type": "cosign container image signature"
+    },
+    "optional": null
+}
+```
+
+#### store "image" in the registry
+
+Let's see how the "image" is stored:
```

Comment:
Now I do not know it. IMO this case will be handled by the cosign signer rather than the registry. There may be some resident service to monitor this case.